### DISCUSSION TITLE
Create CollisionShape3D and MeshInstance3D from Sprite3D

### DISCUSSION THREAD

**Discussion by yosimba2000** [godotengine/godot-proposals#10365] [3D]
Sprite2D allows you to create both 2d colliders and 2d meshes from a sprite.
The same functionality should be offered for Sprite3D nodes, but as their 3D variants. The meshes would be planar, and the colliders would have 0 thickness.

**Comment by Calinou** [comment]
It should be feasible to achieve this using the CollisionPolygon3D node, which provides a way to create extruded 2D collision shapes in 3D space.

However, a shape with 0 thickness may not be usable in all contexts, so perhaps it should always have some thickness to it (even if it's just 0.1 units by default or so).

Out of curiosity, can you describe some use cases for this?

**Reply by yosimba2000** [reply]
I'm making a game in 3D space that uses sprites that the player can click on. But since many sprites visually overlap, I want their collision shapes to match their bitmap shapes.

I just tried copying Sprite2D's generated CollisionPolygon2D's Polygon data to CollisionPolygon3D's Polygon data, and I think it's fine, but can't confirm 100%.

The CollisionPolygon3D shows the same shape as the CollisionPolygon2D, but my data might be too complex to be functional because I'm getting error ```core/math/geometry_2d.cpp:53 - Convex decomposing failed!```, and my game is failing to detect clicks using that CollisionPolygon3D .

**Reply by Calinou** [reply]
I forgot that CollisionPolygon3D only works with convex polygons, while most sprites will generate a concave polygon. There would need to be a convex decomposition step so that the shape is decomposed into multiple convex shapes.

CollisionPolygon2D automatically performs this convex decomposing behind the scenes.
